torch/_dynamo/exc.py

```diff
@@ -73,6 +74,26 @@ def add_to_stats(self, category="unimplemented"):
         counters[category][self.msg] += 1
 
 
+class UserErrorType(Enum):
+    DYNAMIC_CONTROL_FLOW = auto()
+
+
+# TODO Ideally it should inherit from UnsupportedError
+# because if user code doesn't compile, we know it is not
+# supported. But seperate exception for now as to
+# not pollute the stat collector.
+class UserError(TorchDynamoException):
+    def __init__(self, error_type: UserErrorType, msg):
```

Comment:
Please add comment here explaining UserError (same one that you have in the description).

Reply:
done